.github/workflows/tests.yml

```diff
@@ -1,290 +1,152 @@
-        name: jammy_env-artifact
-    - name: Load jammy_env docker
-      if: ${{ needs.id_repo.outputs.in_base_repo == 'false' }}
-      run: |
-        docker load < jammy_env.tar
-    - name: Build
-      run: |
-        cat<<-EOF > run.sh
-          set -xe
-          export CC=gcc-10
-          export CXX=g++-10
-          export FC=gfortran-10
-          git config --global --add safe.directory /src/vendor/kokkos
-          cmake -DCMAKE_CXX_FLAGS="-Wall -Wextra -pedantic -Werror=vla" -DBUILD_BENCHMARKS=ON -DKokkos_ENABLE_CUDA=ON -DKokkos_ARCH_AMPERE80=ON -DKokkos_ENABLE_LIBDL=OFF -DKokkos_ENABLE_DEPRECATED_CODE_3=OFF -DKokkos_ENABLE_DEPRECATED_CODE_4=OFF -DKokkos_ENABLE_DEPRECATION_WARNINGS=OFF /src
-          make -j 2 VERBOSE=1
-        EOF
-        docker run -v ${PWD}:/src:ro ${{ needs.docker-build-jammy_env.outputs.jammy_image }} bash /src/run.sh
-
-  test-hip:
-    runs-on: ubuntu-latest
-    needs: [docker-build-test_env, id_repo]
-    steps:
+      with: { tool-cache: true, large-packages: false }
     - name: Checkout built branch
       uses: actions/checkout@v3
-      with:
-        submodules: recursive
-    - name: Free Disk Space (Ubuntu)
-      uses: jlumbroso/free-disk-space@v1.2.0
-      with:
-        tool-cache: false
-
-        # all of these default to true, but feel free to set to
-        # "false" if necessary for your workflow
-        android: true
-        dotnet: true
-        haskell: true
-        large-packages: false
-        swap-storage: true
-    - name: Collect test_env docker
-      if: ${{ needs.id_repo.outputs.in_base_repo == 'false' }}
+      with: { submodules: recursive }
+    - name: Collect image artifact
+      if: needs.id_repo.outputs.in_base_repo == 'false'
       uses: actions/download-artifact@v3
-      with:
-        name: test_env-artifact
-    - name: Load test_env docker
-      if: ${{ needs.id_repo.outputs.in_base_repo == 'false' }}
+      with: { name: '${{matrix.image}}-artifact' }
+    - name: Load image artifact into docker
+      if: needs.id_repo.outputs.in_base_repo == 'false'
       run: |
-        docker load < test_env.tar
+        docker load < ${{matrix.image}}.tar
+        rm ${{matrix.image}}.tar
     - name: Build
       run: |
-        cat<<-EOF > run.sh
+        cat<<-'EOF' > run.sh
           set -xe
-          export CC=/opt/rocm/hip/bin/hipcc
-          export CXX=/opt/rocm/hip/bin/hipcc
-          git config --global --add safe.directory /src/vendor/kokkos
-          cmake -DCMAKE_CXX_FLAGS="-Wall -Wextra -pedantic -Werror=vla" -DBUILD_BENCHMARKS=ON -DKokkos_ENABLE_HIP=ON -DKokkos_ARCH_VEGA90A=ON -DCMAKE_PREFIX_PATH=/opt/rocm -DKokkos_ENABLE_DEPRECATED_CODE_3=OFF -DKokkos_ENABLE_DEPRECATED_CODE_4=OFF -DKokkos_ENABLE_DEPRECATION_WARNINGS=OFF /src
-          make -j 2 VERBOSE=1
+          git config --global --add safe.directory '*'
+          case "${{matrix.kokkos_backend}}" in
+          'GPU-CUDA')
+            export CC=gcc-10
+            export CXX=g++-10
```

Comment:
You prefer not to specify the `nvcc_wrapper` in the CUDA case btw ?

Reply:
I didn't change it, that's what we did before

Reply:
I'd say we can change it if better, but I would do that in a separate PR

Reply:
Indeed I am surprised. Well let's keep it, just to let you know that Kokkos will you use `nvcc_wrapper` in the end to compile the project if `Kokkos_ENABLE_CUDA` is `ON`